### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by decepulis**
> I'd suggest [hiding whitespace when inspecting this PR](https://github.com/muxinc/video-course-starter-kit/pull/41/files?diff=unified&w=1)!

## So what's new?

At first, I was like "what if I just updated mux node". And then I was like "oh look at these other mux dependencies that need updating". And then I thought "huh one of these other dependencies requires Node 16, and that's EOL. Guess I should update that too. And.... what if I just updated _everything??_". 

And then I thought "what if I move everything to server components?" but that was a step too far. So I just stuck with updating everything.

> seriously, if you want to reject this PR and do this package by package, I'd understand.

Luckily, each major update required only minor changes!
- `next` 12 -> 14
    - `<Link><a><a/></Link>` -> `<Link></Link>`
    - `next/future/image` -> `next/image`
    - next.config.images.domains -> remotePatterns
- `@mux/mux-node` 6 -> 8
    - `client.Video` -> `client.video`
    - `Mux.Webhooks.verifyHeader` -> `client.webhooks.verifySignature`
- `@mux/mux-player-react` 1 -> 2
    - no changes
- `@mux/mux-uploader-react` beta.0 -> beta.15
    - remove `status` property
- `prisma` 4 -> 5
    - `referentialIntegrity` out of preview
    - this is the only package I'm not really comfortable with, so this is the place I'd be most worried about problems
- `@tanstack/react-query` 4 -> 5
    - functions require object argument
    - `isLoading` -> `isPending`
- `clsx` 1 -> 2
    - move from object syntax to function syntax]
- `next-auth` 4.10 -> 4.24
    -  `unstable_getServerSession` -> `getServerSession`
- `react-markdown` 8 -> 9
    - no changes
- `typescript` 4 -> 5
    - new config
    - surprisingly, no changes??
    - seriously every typescript 5 migration I've made has been painful. why was this so easy?

### CODE REVIEW

**Comment by decepulis** [comment]
By the way. If we wanna work with our friends at Vercel to [update this project in the template gallery](https://vercel.com/templates/next.js/video-course-starter-kit), here are some new images that our killer brand team came up with

Mux + Next JS video course update_1200x600@2x
Mux + Next JS video course update_1200x630@2x

File: .vscode/extensions.json
Comment (decepulis) [inline .vscode/extensions.json:1]: We recommend the extension in our README. Now we recommend it in VSCode, too

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ brew install planetscale/tap/pscale
 Next, authorize the Planetscale CLI with your newly created account by running:
 
 ```
-pscale auth
+pscale auth login
```
Comment (decepulis) [inline README.md:65]: Found this command necessary when I was setting up the project, myself

File: pages/api/webhooks/mux.ts
Comment (decepulis) [inline pages/api/webhooks/mux.ts:1]: I wasn't able to test this part (but it looks right?).

**Review by decepulis — COMMENTED** [review]
just a few notes

**Review by davekiss — APPROVED** [review]
much needed cleanup, thanks @decepulis

**Pull request merged**